Kernel/Output/HTML/Preferences/ColumnFilters.pm

```diff
@@ -61,10 +62,23 @@ sub Run {
 
         # pref update db
         if ( !$Kernel::OM->Get('Kernel::Config')->Get('DemoSystem') ) {
+
+            my %Seen;
+            my @ColumnsUnique;
+            COLUMN:
+            for my $Column ( @{ $Param{GetParam}->{$Key} } ) {
+
+                # Skip duplicates.
+                next COLUMN if $Seen{$Column};
+                $Seen{$Column} = 1;
+
+                push @ColumnsUnique, $Column;
+            }
+
```

Comment:
```suggestion
			# remove duplicates.
            if ( $Param{GetParam}->{$Key} && IsArrayRefWithData( $Param{GetParam}->{$Key} ) ) {
                my %UniqueValues = map { $_ => 1 } @{ $Param{GetParam}->{$Key} };
                @{ $Param{GetParam}->{$Key} } = keys %UniqueValues;
            }
```

Reply:
Hi @pboguslawski

these are my suggestions.
Please do not forget the following:


Line: 15
```
use Kernel::System::VariableCheck qw(:all);
```

Reply:
Hi @dennykorsukewitz,

Your algo does not preserve item order in deduplicated array and order must be preserved in this function. Will only add input type verification in next commit (today).

Reply:
Input type verification added in db23b73f13426a4dbc4948afcb7ae163ed900bba.

Reply:
You are right. 
The order is indeed relevant at this one. 👍🏼